src/core/network/processornetwork.cpp

```diff
@@ -620,12 +609,22 @@ Outport* ProcessorNetwork::getOutport(std::string_view path) const {
 bool ProcessorNetwork::isPropertyInNetwork(Property* prop) const {
     if (auto owner = prop->getOwner()) {
         if (auto processor = owner->getProcessor()) {
-            return processor == util::map_find_or_null(processors_, processor->getIdentifier());
+            return processor == util::map_find_or_null(processors_, processor->getIdentifier(),
+                                                       [](const auto& p) { return p.get(); });
         }
     }
     return false;
 }
 
 InviwoApplication* ProcessorNetwork::getApplication() const { return application_; }
 
+void ProcessorNetwork::assignIdentifierAndName(Processor& processor, std::string_view name) {
+    if (processor.getIdentifier().empty()) {
```

Comment:
Why the empty checks? (not intuitive from the name of the function)

Reply:
We only set this in they were not set in the already. This is how we have done it before... I just put it in a function.